src/ILVerify/src/Verifier.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Reflection;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Resources;
+using System.Text;
+using Internal.IL;
+using Internal.TypeSystem;
+using Internal.TypeSystem.Ecma;
+
+namespace ILVerify
+{
+    public class Verifier
+    {
+        private Lazy<ResourceManager> _stringResourceManager =
+            new Lazy<ResourceManager>(() => new ResourceManager("ILVerify.Resources.Strings", Assembly.GetExecutingAssembly()));
+
+        private SimpleTypeSystemContext _typeSystemContext;
+
+        private static VerificationResult s_noSystemModuleResult = new VerificationResult() { Message = "No system module specified" };
+
+        public Verifier(IResolver resolver)
+        {
+            _typeSystemContext = new SimpleTypeSystemContext(resolver);
+        }
+
+        internal Verifier(SimpleTypeSystemContext context)
+        {
+            _typeSystemContext = context;
+        }
+
+        public void SetSystemModuleName(AssemblyName name)
+        {
+            _typeSystemContext.SetSystemModule(_typeSystemContext.GetModule(_typeSystemContext._resolver.Resolve(name)));
+        }
+
+        public IEnumerable<VerificationResult> Verify(PEReader peReader)
+        {
+            if (peReader == null)
+            {
+                throw new ArgumentNullException(nameof(peReader));
+            }
+
+            if (_typeSystemContext.SystemModule is null)
+            {
+                yield return s_noSystemModuleResult;
+                yield break;
+            }
+
+            IEnumerable<VerificationResult> results;
+            try
+            {
+                EcmaModule module = _typeSystemContext.GetModule(peReader);
+                results = VerifyMethods(module, peReader.GetSimpleName(), module.MetadataReader.MethodDefinitions);
+            }
+            catch (VerifierException e)
+            {
+                results = new[] { new VerificationResult() { Message = e.Message } };
+            }
+
+            foreach (var result in results)
+            {
+                yield return result;
+            }
+        }
+
+        public IEnumerable<VerificationResult> Verify(PEReader peReader, TypeDefinitionHandle typeHandle)
+        {
+            if (peReader is null)
```

Comment:
Some places use `is null`, other places use `== null`. It would be nice to be consistent. Personally, `is null` reads harder to me than `== null`.